membership-attribute-service/app/services/IdentityAuthService.scala

```diff
@@ -1,24 +1,38 @@
 package services
 
 import _root_.play.api.mvc.RequestHeader
-import com.gu.identity.play
-import com.gu.identity.play.{AccessCredentials, AuthenticatedIdUser}
-import configuration.Config
-
-object IdentityAuthService extends AuthenticationService {
-  val playAuthService = new play.AuthenticationService {
-    override val identityKeys = Config.idKeys
-
-    override lazy val authenticatedIdUserProvider =
-      AuthenticatedIdUser.provider(
-        AccessCredentials.Cookies.authProvider(identityKeys),
-        AccessCredentials.Token.authProvider(identityKeys,"membership")
-      )
+import com.gu.identity.{IdapiConfig, play}
+import org.http4s.Uri
+import cats.implicits._
+import com.gu.identity.model.User
+import com.gu.identity.play.IdentityPlayAuthService
+import com.gu.identity.play.IdentityPlayAuthService.UserCredentialsMissingError
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
+import scala.concurrent.{ExecutionContext, Future}
+
+class IdentityAuthService(apiConfig: IdapiConfig)(implicit ec: ExecutionContext) extends AuthenticationService {
+
+  val idApiUrl = Uri.unsafeFromString(apiConfig.url)
+
+  val identityPlayAuthService = IdentityPlayAuthService.unsafeInit(idApiUrl, apiConfig.token, Some("membership"))
+
+  def user(implicit requestHeader: RequestHeader): Future[Option[User]] = {
+    getUser(requestHeader)
+      .map(user => Option(user))
+      .handleError { err =>
+        if(err.isInstanceOf[UserCredentialsMissingError])
+          SafeLogger.info(s"unable to authorize user - $err")
+        else
+          SafeLogger.error(scrub"unable to authorize user - $err")
```

Comment:
is "err" an exception? very minor, but if so I think you can pass it to the second parameter of `SafeLogger.error` in order to get the stack trace in the logs

Reply:
Good point. Will do.